src/amr/data/field/refine/electric_field_refiner.hpp

```diff
@@ -0,0 +1,299 @@
+#ifndef PHARE_ELECTRIC_FIELD_REFINER_HPP
+#define PHARE_ELECTRIC_FIELD_REFINER_HPP
+
+#include <SAMRAI/hier/Box.h>
+
+#include "amr/resources_manager/amr_utils.hpp"
+#include "core/utilities/constants.hpp"
+#include "core/data/grid/gridlayoutdefs.hpp"
+#include "core/utilities/point/point.hpp"
+#include "core/utilities/constants.hpp"
+
+#include <cstddef>
+
+namespace PHARE::amr
+{
+
+using core::dirX;
+using core::dirY;
+using core::dirZ;
+
+template<std::size_t dimension>
+class ElectricFieldRefiner
+{
+public:
+    ElectricFieldRefiner(std::array<core::QtyCentering, dimension> const& centering,
+                         SAMRAI::hier::Box const& destinationGhostBox,
+                         SAMRAI::hier::Box const& sourceGhostBox,
+                         SAMRAI::hier::IntVector const& ratio)
+        : fineBox_{destinationGhostBox}
+        , coarseBox_{sourceGhostBox}
+        , centerings_{centering}
+    {
+    }
+
+
+    /** @brief Given a sourceField , a destinationField, and a fineIndex compute the
+     */
+    template<typename FieldT>
+    void operator()(FieldT const& coarseField, FieldT& fineField,
+                    core::Point<int, dimension> fineIndex)
+    {
+        TBOX_ASSERT(coarseField.physicalQuantity() == fineField.physicalQuantity());
+
+        auto locFineIdx = AMRToLocal(fineIndex, fineBox_);
+        auto coarseIdx{fineIndex};
+        for (auto& idx : coarseIdx)
+            idx = idx / 2;
+        auto locCoarseIdx = AMRToLocal(coarseIdx, coarseBox_);
+
+
+        if constexpr (dimension == 1)
```

Comment:
would it be feasible to have dim specific functions?

Reply:
done